core/unit_test/TestAtomics.hpp

```diff
@@ -539,4 +539,45 @@ TEST(TEST_CATEGORY, atomics) {
 #endif
 }
 
+// see https://github.com/trilinos/Trilinos/pull/11506
+struct TpetraUseCase {
+  template <class Scalar>
+  struct WrapScalarAndCompareAbsMax {
+    Scalar value;
+
+   private:
+    friend KOKKOS_FUNCTION bool operator<(
+        WrapScalarAndCompareAbsMax const& lhs,
+        WrapScalarAndCompareAbsMax const& rhs) {
+      return Kokkos::abs(lhs.value) < Kokkos::abs(rhs.value);
+    }
+    friend KOKKOS_FUNCTION bool operator>(
+        WrapScalarAndCompareAbsMax const& lhs,
+        WrapScalarAndCompareAbsMax const& rhs) {
+      return Kokkos::abs(lhs.value) > Kokkos::abs(rhs.value);
+    }
+  };
+
+  using T = int;
+  Kokkos::View<T> d_{"lbl"};
+  KOKKOS_FUNCTION void operator()(int i) const {
+    // 0, -1, 2, -3, ...
+    auto v_i = static_cast<T>(i);
+    if (i % 2 == 1) v_i = -v_i;
+    Kokkos::atomic_max(reinterpret_cast<WrapScalarAndCompareAbsMax<T>*>(&d_()),
+                       WrapScalarAndCompareAbsMax<T>{v_i});
+  }
+  TpetraUseCase() {
+    Kokkos::deep_copy(d_, Kokkos::Experimental::finite_min_v<T>);
```

Comment:
```suggestion
    Kokkos::deep_copy(d_, 0);
```
or
```suggestion
```
Otherwise, the initial value dominates.

Reply:
No I meant it, I am doing "max", this is the right identity value.

Reply:
The CI proves you wrong. 🙂 For integer types `finite_min` is `INT_MIN` (min and lowest are the same).

Reply:
oh right...  ty